### PR TITLE
Optimize tp broadcast

### PR DESCRIPTION
**Pull request by grimoire**
requirement:

- [x] #2810

### CODE REVIEW

File: lmdeploy/pytorch/model_inputs.py [hunk 1]
```diff
@@ -36,10 +45,45 @@ def to_device(self, device: str):
...
                         data = [d.to_device(device) for d in data]
-                        mm_datas[modal_type] = data
+                        new_mm_datas[modal_type] = data
+                    new_v.append(new_mm_datas)
+                v = new_v
+            out_dict[k] = v
+
+        return VisionModelInputs(**out_dict)
+
+    def broadcast(self):
```
Comment (AllentDan) [inline lmdeploy/pytorch/model_inputs.py:60]: How can we test the feature? Just using tp mode for a VL model?

Comment (grimoire) [reply lmdeploy/pytorch/model_inputs.py:60]: Yes

**Review by AllentDan — APPROVED** [review]
Tested OK with Qwen2-VL

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by lvhan028** [comment]
Will it affect backends supported by dlinfer?

**Pull request merged**